dm/loader/lightning.go

```diff
@@ -234,15 +235,15 @@ func (l *LightningLoader) restore(ctx context.Context) error {
 		}
 		cfg.Routes = l.cfg.RouteRules
 
-		// TODO lightning checkpoint will support s3 in other pr
-		if storage.IsS3Path(l.cfg.LoaderConfig.Dir) {
-			cfg.Checkpoint.Enable = false
-		} else {
-			cfg.Checkpoint.Driver = lcfg.CheckpointDriverFile
-			cpPath := filepath.Join(l.cfg.LoaderConfig.Dir, lightningCheckpointFileName)
-			cfg.Checkpoint.DSN = cpPath
-			cfg.Checkpoint.KeepAfterSuccess = lcfg.CheckpointOrigin
+		cfg.Checkpoint.Driver = lcfg.CheckpointDriverFile
+		var cpPath string
+		cpPath, err = storage.AdjustPath(l.cfg.LoaderConfig.Dir, string(filepath.Separator)+lightningCheckpointFileName)
```

Comment:
found that in SubTaskConfig.Adjust, we already make sure `l.cfg.LoaderConfig.Dir` is unique.

Reply:
yes, when add suffix to `l.cfg.LoaderConfig.Dir` to construct a file for `cfg.Checkpoint.DSN`, we actually don't need to check unique, because `l.cfg.LoaderConfig.Dir`'s suffix is 'taskid_sourceid' and `lightningCheckpointFileName` is `tidb_lightning_checkpoint.pb`, but if we check unique may be ok and it will make `storage.AdjustPath`'s  logic be pithier.

Reply:
`cpPath := filepath.Join(l.cfg.LoaderConfig.Dir, lightningCheckpointFileName)` won't cause any problem and your change introduced https://github.com/pingcap/tiflow/pull/4786#discussion_r825559207

Reply:
seems we can keep `cpPath := filepath.Join(l.cfg.LoaderConfig.Dir, lightningCheckpointFileName)`

Reply:
Use `storage.AdjustPath` is because here want to use Lightning checkpoint support S3 in this [PR](https://github.com/pingcap/tidb/pull/32373)

Reply:
i add some notes to reduce ambiguity.

Reply:
@WizardXiao I still can't understand why we need `AdjustPath` again. We have made sure `l.cfg.LoaderConfig.Dir` is unique in SubTaskConfig.Adjust

Reply:
this is Lightning checkpoint need a fileName, it used to be `l.cfg.LoaderConfig.Dir/tidb_lightning_checkpoint.pb` and `AdjustPath` for this time use to add  `/tidb_lightning_checkpoint.pb` and make a new path, but this will not change `l.cfg.LoaderConfig.Dir`. Maybe this function name is easy to make ambiguity.

Reply:
> this is because Lightning checkpoint need a fileName

why can't use `cpPath := filepath.Join(l.cfg.LoaderConfig.Dir, lightningCheckpointFileName)`

Reply:
this is because `l.cfg.LoaderConfig.Dir` may be a s3 path.